promotion/controllers.go

```diff
@@ -213,6 +216,89 @@ func ClaimPromotion(service *Service) handlers.AppHandler {
 	})
 }
 
+// OrderItemRequest is a thing
+type OrderItemRequest struct {
+	SKU     string `json:"sku"`
+	Quanity int    `json:"quanity"`
+}
+
+// CreateOrderRequest includes information needed to create a promotion
+type CreateOrderRequest struct {
+	Items []OrderItemRequest `json:"items"`
+}
+
+// CreateOrder is the handler for creating a new order
+func CreateOrder(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		var req CreateOrderRequest
+		err := requestutils.ReadJSON(r.Body, &req)
+		if err != nil {
+			return handlers.WrapError(err, "Error in request body", http.StatusBadRequest)
+		}
+
+		_, err = govalidator.ValidateStruct(req)
+		if err != nil {
+			return handlers.WrapValidationError(err)
+		}
+
+		totalPrice := decimal.New(0, 0)
+		orderItems := []OrderItem{}
+		for i := 0; i < len(req.Items); i++ {
+			orderItem := CreateOrderItemFromMacaroon(req.Items[i].SKU, req.Items[i].Quanity)
+			totalPrice = totalPrice.Add(orderItem.Subtotal)
+
+			orderItems = append(orderItems, orderItem)
+		}
+
+		order, err := service.datastore.CreateOrder(totalPrice, "brave.com", "pending", orderItems)
+		if err != nil {
+			panic(err)
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		if err := json.NewEncoder(w).Encode(order); err != nil {
+			panic(err)
+		}
+		return nil
+	})
+}
+
+// GetOrder is the handler for creating a new order
```

Comment:
is it? :)

Reply:
😂 copy / paste err 🍝

https://github.com/brave-intl/bat-go/pull/226/commits/1429355a01dffd43e8b5809ead3228a59cf77317